connmgr/connmanager_test.go

```diff
@@ -39,6 +40,9 @@ type mockConn struct {
 
 	// remote network, address for the connection.
 	rAddr net.Addr
+
+	// context for the connection.
+	ctx context.Context
```

Comment:
Same here.  The context shouldn't be stored, rather it should be passed explicitly to where it is needed.